lib/executor.js

```diff
@@ -330,24 +332,37 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
-    data.config = getUpdatedConfigObject(app, data.config);
+    var useEnvVar = true;
+    data.config = getUpdatedConfigObject(app, data.config, useEnvVar);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
-function getUpdatedConfigObject(app, config) {
+function getUpdatedConfigObject(app, config, useEnvVar) {
   var DYNAMIC_CONFIG_PARAM = /\$\{(\w+)\}$/;
 
   function getConfigVariable(param) {
     var configVariable = param;
     var match = configVariable.match(DYNAMIC_CONFIG_PARAM);
     if (match) {
-      var appValue = app.get(match[1]);
-      if (appValue !== undefined) {
+      if (useEnvVar && process.env[match[1]] !== undefined) {
+        debug('Dynamic Configuration: Using process.env: ' +
+          param, process.env[match[1]]);
+        configVariable = process.env[match[1]];
+      } else if (app.get(match[1]) !== undefined) {
+        debug('Dynamic Configuration: Using app.get(server/config.json): ' +
+          param, app.get(match[1]));
+        var appValue = app.get(match[1]);
         configVariable = appValue;
       } else {
         console.warn('%s does not resolve to a valid value. ' +
-        '"%s" must be resolvable by app.get().', param, match[1]);
+        '"%s" must be resolvable in Environment variable or by app.get().',
```

Comment:
Please improve the message to tell the user what value will be used and also include the parameter name & variable name.

```js
console.warn('%s does not resolve to a valid value, using "undefined" instead. ' +
  // ...
```